pkg/controller/certificates/util.go

```diff
@@ -122,12 +122,14 @@ func certificateMatchesSpec(crt *v1alpha2.Certificate, key crypto.Signer, cert *
 	}
 
 	// validate that the issuer is correct
-	if crt.Spec.IssuerRef.Name != secret.Annotations[v1alpha2.IssuerNameAnnotationKey] {
+	if crt.Spec.IssuerRef.Name != secret.Annotations[v1alpha2.IssuerNameAnnotationKey] &&
+		crt.Spec.IssuerRef.Name != secret.Annotations[v1alpha2.DeprecatedIssuerNameAnnotationKey] {
```

Comment:
This doesn't account for the case where the *deprecated* value is "" and the *new* value is correct - in which case, we *shouldn't* re-issue.

It also doesn't account for the case where the case where the *old* value is incorrect, but the *new* value _is_ correct, unless I am mis-understanding (and there should be corresponding unit tests for this)